### PR TITLE
[Google Blockly] Add accessibility plugin (Keyboard navigation)

### PR DESCRIPTION
**Pull request by mikeharv**
This adds Blockly plugin for keyboard navigation in our Google Blockly labs. This allows users to use the keyboard to navigate the blocks in the workspace and toolbox. 

> A user can enable and disable keyboard navigation by hitting Shift + Ctrl + k. When keyboard navigation is first enabled a flashing red line appears on the workspace. This is the cursor. It shows the user's current location and updates as the user navigates the workspace. [[Full documentation](https://developers.google.com/blockly/guides/configure/web/keyboard-nav)]

We can also add an option to toggle modes in the workspace's context menu.

This video demonstrates writing a simple Blockly program for Dance Party without using the mouse in about one minute. *Note that the keystroke monitoring on the left comes from [keycastr](https://github.com/keycastr/keycastr). It is just there for illustration purposes and is not part of this pull request.*

https://user-images.githubusercontent.com/43474485/211643222-cf5a1f76-4308-472e-95fb-6390c34b1c2f.mp4

## Links

Plugin: https://github.com/google/blockly-samples/tree/master/plugins/keyboard-navigation
- Documentation: [Keyboard Navigation  |  Blockly  |  Google Developers](https://developers.google.com/blockly/guides/configure/web/keyboard-nav)
- jira ticket: [SL-423: [Timebox: 1 week] Prototype Google Blockly accessibility mode](https://codedotorg.atlassian.net/browse/SL-423)

## Deployment strategy

Note: This feature branch includes the Blockly 9 upgrade cannot be merged until our SauceLabs powered UI tests move from Safari 12 to Safari 13.

## Follow-up work

Work with Product to determine how we want to support users who are interested in trying this new feature. (Support doc, etc.)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/yarn.lock [hunk 1]
```diff
@@ -1306,6 +1306,15 @@
   version "2.0.2"
   resolved "https://registry.yarnpkg.com/@blockly/field-grid-dropdown/-/field-grid-dropdown-2.0.2.tgz#61c531e4082f0424a8afccd9c87ebe5286744158"
   integrity sha512-wA0nTdhgNI4xGPcODOuQhB43LcvDtwWZcorlNtyuTFJcYHidYOyTkC5i99HpiUxo4qdrt0C3OqkLp1gmJPEBXA==
+"@blockly/field-grid-dropdown@~2.0.2":
```
Comment (maddiedierker) [inline apps/yarn.lock]: i'm confused about this change -- we already have `@blockly/field-grid-dropdown@~2.0.2` directly above this line, and the information looks identical

Comment (mikeharv) [reply apps/yarn.lock]: Nice catch - this was likely just a mistake on my part due to a merge conflict. i must have accepted both versions here.

File: apps/src/blockly/addons/contextMenu.js [hunk 2]
```diff
@@ -184,6 +185,36 @@ const registerUnshadow = function() {
   GoogleBlockly.ContextMenuRegistry.registry.register(unshadowOption);
 };
 
+const registerKeyboardNavigation = function() {
+  const keyboardNavigationOption = {
+    displayText: function(scope) {
+      const displayText = scope.workspace.keyboardAccessibilityMode
+        ? msg.blocklyKBNavOff()
+        : msg.blocklyKBNavOn();
+      return displayText;
```
Comment (maddiedierker) [inline apps/src/blockly/addons/contextMenu.js]: nit: the `displayText` variable feels unnecessary here -- returning the value from the ternary here is probably clear enough

File: apps/src/blockly/addons/contextMenu.js [hunk 3]
```diff
@@ -184,6 +185,36 @@ const registerUnshadow = function() {
...
+        console.log('navigationController not found');
+        return 'hidden';
+      }
+    },
+    callback: function(scope) {
+      if (scope.workspace.keyboardAccessibilityMode) {
+        Blockly.navigationController.disable(scope.workspace);
+      } else {
+        Blockly.navigationController.enable(scope.workspace);
+      }
```
Comment (maddiedierker) [inline apps/src/blockly/addons/contextMenu.js]: super nit / subjective comment: a ternary might be clearer here

**Review by maddiedierker — APPROVED** [review]
left some very minor nit comments -- neither are merge-blockers. this looks great, it's so wonderful to see this plugin enabled for google blockly! 🎉

File: apps/src/blockly/addons/contextMenu.js (hunk 2 above)
Comment (mikeharv) [reply apps/src/blockly/addons/contextMenu.js]: That's fair. There are other context menu options in this file with more complex values where the use of the variable was justified. I think I primarily kept it here to maintain a pattern. No need though if it just reads as more complicated!

File: apps/src/blockly/addons/contextMenu.js (hunk 3 above)
Comment (mikeharv) [reply apps/src/blockly/addons/contextMenu.js]: Oh, good call. Believe it or not, I think I've only ever used ternary to produce a value (like `displayText` above), and hadn't ever considered using them to call functions like these that don't return anything.

File: apps/src/blockly/addons/contextMenu.js [hunk 4]
```diff
@@ -184,6 +185,36 @@ const registerUnshadow = function() {
...
+      const displayText = scope.workspace.keyboardAccessibilityMode
+        ? msg.blocklyKBNavOff()
+        : msg.blocklyKBNavOn();
+      return displayText;
+    },
+    preconditionFn: function() {
+      if (Blockly.navigationController) {
+        return 'enabled';
+      } else {
+        console.log('navigationController not found');
```
Comment (molly-moen) [inline apps/src/blockly/addons/contextMenu.js]: should this log be here?

**Review by molly-moen — APPROVED** [review]
LGTM!

File: apps/src/blockly/addons/contextMenu.js (hunk 4 above)
Comment (mikeharv) [reply apps/src/blockly/addons/contextMenu.js]: Thanks! After removing this log, I was able to compress 5 lines down to 1 by using a ternary.:)

**Review by fisher-alice — APPROVED** [review]
Yay! This is so exciting. Just wanted to log the blockly samples [bug report](https://github.com/google/blockly-samples/issues/1453) submitted for this plugin when the 'zelos' renderer is selected. Thanks for taking the lead on this.

**Pull request merged**